mocks/firebase.js

```diff
@@ -21,10 +21,19 @@ const firebaseStub = overrides => {
 };
 
 const mockFirebase = (overrides = {}) => {
-  jest.mock('firebase', () => firebaseStub(overrides)) &&
-    jest.mock('firebase-admin', () => firebaseStub(overrides));
+  mockModuleIfFound('firebase', overrides);
+  mockModuleIfFound('firebase-admin', overrides);
 };
 
+function mockModuleIfFound(moduleName, overrides) {
```

Comment:
Not sure I follow how this function is being used.  Is it necessary? If so, could you document it in the README perhaps?

Reply:
Sorry, maybe my commit description was too cryptic here.

The point is, that if this package is used in a project where you do not have the `firebase` dependency, but only `firebase-admin` it fails on this line `jest.doMock('firebase', () => firebaseStub(overrides));`, because `jest` can not find the `firebase` module.

This change applies mocking only if a respective module can be found as a dependency - `require.resolve()` was the best way I found to check if a package can be found.

The function is not exported, so I don't think it's worth describing it in the README.

In addition, I changed `mock` functions to `doMock` - when plain mocking was replaced with the function call the `&&` trick did not work anymore and `babel-plugin-jest-hoist` reported an error: `The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables`.

Reply:
Ah, thanks for explaining. Makes sense